Apps/WebClient/src/ClientApp/src/components/report/medicationHistory.vue

```diff
@@ -130,29 +121,27 @@ export default class MedicationHistoryReportComponent extends Vue {
     }
 
     public async generatePdf(): Promise<void> {
-        this.logger.debug("generating Medication History PDF...");
-        this.totalFiles = Math.ceil(
-            this.visibleRecords.length / this.fileMaxRecords
+        const reportService: IReportService = container.get<IReportService>(
+            SERVICE_IDENTIFIER.ReportService
         );
-        this.isPreview = false;
-
-        return this.generatePdfFile().then(() => {
-            if (this.fileIndex + 1 < this.totalFiles) {
-                this.fileIndex++;
-                return this.generatePdfFile();
-            } else {
-                this.isPreview = true;
-                this.fileIndex = 0;
-            }
-        });
-    }
 
-    private generatePdfFile(): Promise<void> {
-        return PDFUtil.generatePdf(
-            `HealthGateway_MedicationHistory_File${this.fileIndex + 1}.pdf`,
-            this.report,
-            `File ${this.fileIndex + 1} of ${this.totalFiles}`
-        );
+        return reportService
+            .generateReport({
+                data: {
+                    patientData: this.patientData,
+                    medicationStatements: this.visibleRecords,
```

Comment:
This will need to be updated to match the JSON expected in the template (or vice versa).  Sample JSON is attached to the AB#10773 ticket.

Reply:
updated

Reply:
notice i updated the templated to match the new data structure